### PR TITLE
Check if reporting Set fits gdx

### PR DESCRIPTION
**Pull request by robinhasse**
The reporting template is now checked for consistency with the gdx. Surplus set elements trigger a warning and missing elements an error. This should help the user when selecting a reporting template that doesn't fit the run.

As bs and hs are currently just copies of bsr and hsr respectively, you always get a warning for the 0 elements. Not sure if we should change that such that or live with an expected warning. I could either implement an exception for the warning or the creation of aliases in the mapping.

### CODE REVIEW

File: R/convGDX2MIF.R [hunk 1]
```diff
@@ -27,6 +32,17 @@ convGDX2MIF <- function(gdx, tmpl = NULL, file = NULL, scenario = "default", t =
   }
 
   brickSets <- readBrickSets(tmpl)
+  setElements <- .findInconsistenSetElements(brickSets, gdx)
+  if (!is.null(setElements[["surplus"]])) {
+    warning("The following set elements are mapped in the reporting template ",
+            "but not found in the gdx:\n  ",
+            paste(capture.output(setElements[["surplus"]]), collapse = "\n  "))
+  }
```
[the remark this answers is not in the document]
Comment (robinhasse) [reply R/convGDX2MIF.R]: Your are right, I was too lazy. In the reporting template, `bs` and `hs` are no longer defined as aliases of `bsr` and `hsr` respectively but as childs. As such, they can contain less elements and subsets than their parent set. This allows to make the test for inconsistencies more stringent. Any inconsistency (missing or surplus elements) between reporting template and gdx (missing or surplus elements) will lead to an error now.

File: R/convGDX2MIF.R [hunk 2]
```diff
@@ -27,6 +35,11 @@ convGDX2MIF <- function(gdx, tmpl = NULL, file = NULL, scenario = "default", t =
   }
 
   brickSets <- readBrickSets(tmpl)
+  inconsistencies <- .findInconsistenSetElements(brickSets, gdx)
+  if (!is.null(inconsistencies)) {
+    stop("The reporting template is not consisten with the gdx:\n  ",
```
Comment (ricardarosemann) [inline R/convGDX2MIF.R:40]:
```suggestion
    stop("The reporting template is not consistent with the gdx:\n  ",
```
Spelling error

**Review by ricardarosemann — APPROVED** [review]
Thank you for the change.

I found one spelling error, all else is good.

File: R/convGDX2MIF.R (hunk 2 above)
Comment (robinhasse) [reply R/convGDX2MIF.R]: Thanks for spotting. I corrected that.

**Pull request merged**